pkg/api/handlers/libpod/volumes.go

```diff
@@ -113,25 +118,30 @@ func ListVolumes(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if len(query.Filters) > 0 {
-		volumeFilters, err = generateVolumeFilters(query.Filters)
-		if err != nil {
-			utils.InternalServerError(w, err)
-			return
-		}
+	volumeFilters, err := filters.GenerateVolumeFilters(query.Filters)
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
 	}
+
 	vols, err := runtime.Volumes(volumeFilters...)
 	if err != nil {
 		utils.InternalServerError(w, err)
 		return
 	}
 	for _, v := range vols {
-		config, err := v.Config()
-		if err != nil {
-			utils.InternalServerError(w, err)
-			return
+		config := entities.VolumeConfigResponse{
+			Name:       v.Name(),
+			Driver:     v.Driver(),
+			Mountpoint: v.MountPoint(),
+			CreatedAt:  v.CreatedTime(),
+			Labels:     v.Labels(),
+			Scope:      v.Scope(),
+			Options:    v.Options(),
+			UID:        v.UID(),
+			GID:        v.GID(),
 		}
-		volumeConfigs = append(volumeConfigs, config)
+		volumeConfigs = append(volumeConfigs, &entities.VolumeListReport{VolumeConfigResponse: config})
```

Comment:
I don't think we want entities to leak into here - struct should live in pkg/api and entities should use it

Reply:
we agreed to this on Friday ... imho, this makes more sense as is.  @jwhonce what say you?

Reply:
I'm with @baude entities, should be used in all the "presentation" layers.